apple/swift-docc
# Use a more detailed page type for C++ namespaces in navigator index

Bug/issue #, if applicable: 122340677

## Summary

This adds a new `namespace` case to the `NavigatorIndex.PageType` enum that can be used in `RenderIndex` to emit a more specific `"namespace"` page type instead of a generic `"symbol"` type in the Render Index JSON.

This is needed so that Swift-DocC-Render can distinguish namespaces from other generic symbols in the navigator and render them appropriately, like choosing an icon that is specific to namespaces.

## Testing

Steps:
1. Run `swift run docc convert Tests/SwiftDocCTests/Test\ Bundles/CxxSymbols.docc`
2. Verify that the node for the `Foo` namespace symbol in `Tests/SwiftDocCTests/Test\ Bundles/CxxSymbols.docc/.docc-build/index/index.json` now has a `type` value of `"namespace"` and not `"symbol"`

## Checklist

Make sure you check off the following items. If they cannot be completed, provide a reason.
